mainwindow.cpp

```diff
@@ -577,78 +513,96 @@ QStringList MainWindow::getAttackParameters()
     } else {
         // In case we do not have explicit format we erase remembered
         // key.
-        m_format = "";
+        m_format.clear();
     }
+    m_settings.setValue("formatJohn", m_format);
+    m_settings.setValue("formatUI", m_ui->comboBox_Format->currentText());
```

Comment:
But I can modify it to store only one and redo the special treatment if you want